jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -172,15 +172,19 @@ static void afterClinitInitialization() {
 		/*[PR JAZZ 58297] - continue with the rules defined by JAZZ 57070 - Build a Java 8 J9 JCL using the SIDECAR18 preprocessor configuration */
 		// Check the default encoding
 		/*[Bug 102075] J2SE Setting -Dfile.encoding=junk fails to run*/
-		/*[IF Sidecar19-SE]*/
+		/*[IF !Sidecar19-SE]*/
+		StringCoding.encode(new char[1], 0, 1);
```

Comment:
Does moving this effect initialization order?  We've had issues with initializing string encoding-related functions at the wrong time....

Reply:
The resulting post processed code is the same in all cases.

Reply:
From code history, `StringCoding.encode` was invoked before `setErr` & `setOut` for both Java 8 & 9 around five weeks ago. `StringCoding.encode` was moved after `setErr` & `setOut` unintentionally in a changeset for this code area. I suggest to put `StringCoding.encode` for `Java 9` back to initial location in same order as `Java 8` as following code snippet:
		/\*[IF Sidecar19-SE]\*/
		StringCoding.encode(String.LATIN1, new byte[1]);
		/\*[ELSE]\*/
		StringCoding.encode(new char[1], 0, 1);
		/\*[ENDIF]\*/
		/\*[IF Sidecar19-SE|Sidecar18-SE-OpenJ9]\*/
		setErr(new PrintStream(new BufferedOutputStream(new FileOutputStream(FileDescriptor.err)), true));
		setOut(new PrintStream(new BufferedOutputStream(new FileOutputStream(FileDescriptor.out)), true));
		/\*[IF Sidecar19-SE_RAWPLUSJ9]\*/
		setIn(new BufferedInputStream(new FileInputStream(FileDescriptor.in)));
		/\*[ENDIF]\*/
		/\*[ELSE]\*/
		/\*[PR s66168] - ConsoleInputStream initialization may write to System.err \*/
		/\*[PR s73550, s74314] ConsolePrintStream incorrectly initialized \*/
		setErr(com.ibm.jvm.io.ConsolePrintStream.localize(new BufferedOutputStream(new FileOutputStream(FileDescriptor.err)), true));
		setOut(com.ibm.jvm.io.ConsolePrintStream.localize(new BufferedOutputStream(new FileOutputStream(FileDescriptor.out)), true));
		/\*[ENDIF]\*/